web/app/mu-plugins/filters.php

```diff
@@ -172,15 +200,31 @@ function __is_login_url( string $url ) : bool {
  * @return string The normalized URL.
  */
 function __normalize_wp_url( string $url ): string {
-    $scheme = parse_url( $url, PHP_URL_SCHEME );
-    $scheme_with_separator = $scheme ? $scheme . '://' : '';
+	// Parse the URL into components.
+	$parts = parse_url( $url );
 
-    // Remove the scheme from the URL if it exists.
-    $remaining_url = $scheme ? substr( $url, strlen($scheme_with_separator ) ) : $url;
+	// Normalize the URL to remove any double slashes.
+	if ( isset( $parts['path'] ) ) {
+		$parts['path'] = preg_replace( '#/+#', '/', $parts['path'] );
+	}
 
-    // Normalize the remaining URL to remove any double slashes.
-    $normalized_url = str_replace( '//', '/', $remaining_url );
+	// Rebuild and return the full normalized URL.
+	return __rebuild_url_from_parts( $parts );
+}
```

Comment:
seems over-engineered with the same result but maybe I'm missing what's changed and how the bugs manifested.

Reply:
I'm not entirely convinced it's necessary but I'm also not convinced that it's not. It was one of the many steps on the path of resolving the bug(s).